File: app/src/main/java/com/github/android/lvrn/lvrnproject/view/fragment/entitieslist/EntitiesListWithSearchPresenter.java
package com.github.android.lvrn.lvrnproject.view.fragment.entitieslist;

import androidx.core.view.MenuItemCompat;
import android.view.MenuItem;

import com.github.android.lvrn.lvrnproject.service.form.ProfileDependedForm;
import com.github.valhallalabs.laverna.persistent.entity.base.ProfileDependedEntity;

/**
 * @author Vadim Boitsov <vadimboitsov1@gmail.com>
 */

public interface EntitiesListWithSearchPresenter<T1 extends ProfileDependedEntity, T2 extends ProfileDependedForm<T1>> extends EntitiesListPresenter<T1, T2> , MenuItemCompat.OnActionExpandListener{

    void disposeSearch();

    void subscribeSearchView(MenuItem searchItem);
}

File: app/src/main/java/com/github/android/lvrn/lvrnproject/view/fragment/entitieslist/EntitiesListWithSearchPresenter.kt
package com.github.android.lvrn.lvrnproject.view.fragment.entitieslist

import android.view.MenuItem
import androidx.core.view.MenuItemCompat
import com.github.android.lvrn.lvrnproject.service.form.ProfileDependedForm
import com.github.valhallalabs.laverna.persistent.entity.base.ProfileDependedEntity

/**
 * @author Vadim Boitsov <vadimboitsov1></vadimboitsov1>@gmail.com>
 */
interface EntitiesListWithSearchPresenter<T1 : ProfileDependedEntity, T2 : ProfileDependedForm<T1>> :
    EntitiesListPresenter<T1, T2>, MenuItemCompat.OnActionExpandListener {
    fun disposeSearch()

    fun subscribeSearchView(searchItem: MenuItem?)
}
